desktop/src/Desktop/Frontend.hs

```diff
@@ -92,15 +99,62 @@ fileStorage dir = Storage
     where path :: Show a => a -> FilePath
           path k = dir </> FilePath.makeValid (show k)
 
-bipCrypto :: Crypto.XPrv -> Text -> Crypto Crypto.XPrv
+newtype XPactSecret = XPactSecret { unXPactSecret :: BS.ByteString }
+
+instance Aeson.ToJSON XPactSecret where
+  toJSON = Aeson.toJSON . T.decodeUtf8 . B16.encode . unXPactSecret
+
+instance Aeson.FromJSON XPactSecret where
+  -- TODO: Should there be some verification of length like XPrv?
+  parseJSON = fmap (XPactSecret . fst . B16.decode . T.encodeUtf8) . Aeson.parseJSON
+
+data DesktopKey
+  = DesktopKeyBip32 Crypto.XPrv
+  | DesktopKeyPact PPKScheme PublicKey XPactSecret
+  deriving (Generic)
+
+-- TODO: This is not backwards compatible. Do we care?
+instance Aeson.FromJSON DesktopKey
+instance Aeson.ToJSON DesktopKey
+
+-- The pact 3.0 blog said that pact supports ethereum ECDSA and ED25519 keys. We probably need
+-- to support both here. See https://medium.com/kadena-io/announcing-pact-3-0-4b0a8f35e6a0
+bipCrypto :: Crypto.XPrv -> Text -> Crypto DesktopKey
```

Comment:
Not for now, but we should drop in an asana ticket to maybe separate the type for storage vs signing. @tomsmalley  suggested maybe doing an associated type in the crypto class and that sounds pretty good to me.

Reply:
Ticket created